### PR TITLE
Add netplan config parding sh utility for ubuntu static ip

### PR DESCRIPTION
**Pull request by yaacov**
Add netplan config parding sh utility for ubuntu static ip

### CODE REVIEW

**Review by fabiand — CHANGES_REQUESTED** [review]
GOLD for bash/dash yaml parsing.

But let's eval to use `netplan generate` or `netplan get` to avoid this parsing.

**Comment by fabiand** [comment]
With the following script we loop over all configured interfaces and output the one for the given IP address

```console
IP_TO_FIND=10.0.2.15
netplan get ethernets | grep -Eo "^[^[:space:]]+[^:]" | while read IFNAME  ; do netplan get ethernets.$IFNAME.addresses | grep -q $IP_TO_FIND && echo $IFNAME ; done
```

File: virt-v2v/scripts/rhel/run/network_config_util.sh [hunk 1]
```diff
@@ -24,9 +28,19 @@ if [ -f "$UDEV_RULES_FILE" ]; then
...
 fi
 
+# Helper functions
+# ----------------
+
+# Get the v2v map file as list 
+v2v_list_as_envs() {
+    while IFS= read -r line; do
+        echo "$line"
+    done < "$V2V_MAP_FILE"
```
Comment (fabiand) [inline virt-v2v/scripts/rhel/run/network_config_util.sh:38]: I think this is equal to `cat $V2V_MAP_FILE` as we just echo every line without modifications.

In the original code I was modifying the line to convert them into ENV_KEY=VALUE format to then export them iwth `export`

File: virt-v2v/scripts/rhel/run/network_config_util.sh [hunk 2]
```diff
@@ -104,9 +123,61 @@ udev_from_nm() {
...
+    }
+
+    # Loop over all interface names and treturn the one with target_ip, or null
+    find_interface_by_ip() {
+        target_ip="$1"
+
+        # Loop through all interfaces and check for the given IP address
+        netplan_get ethernets | grep -Eo "^[^[:space:]]+[^:]" | while read IFNAME; do
+            if netplan_get ethernets."$IFNAME".addresses | grep -q "$target_ip"; then
+                echo "$IFNAME"
```
Comment (fabiand) [inline virt-v2v/scripts/rhel/run/network_config_util.sh:149]: either we should `return` here
or later we need to check that we only found one ifname

File: virt-v2v/scripts/rhel/run/network_config_util.sh [hunk 3]
```diff
@@ -104,9 +123,61 @@ udev_from_nm() {
...
+
+    # Read the mapping file line by line
+    v2v_list_as_envs | while read line;
+    do
+        # Extract S_HW and S_IP from the current line in the mapping file
+        extract_mac_ip "$line"
+
+        # If S_HW and S_IP were not extracted, skip the line
+        if [ -z "$S_HW" ] || [ -z "$S_IP" ]; then
+            continue
```
Comment (fabiand) [inline virt-v2v/scripts/rhel/run/network_config_util.sh:162]: worth adding a log message?

File: virt-v2v/scripts/rhel/run/network_config_util.sh [hunk 4]
```diff
@@ -104,9 +123,61 @@ udev_from_nm() {
...
+        fi
+
+        # Search the parsed netplan output for a matching IP address
+        interface_name=$(find_interface_by_ip "$S_IP")
+
+        # If no interface is found, skip this entry
+        if [ -z "$interface_name" ]; then
+            continue
```
Comment (fabiand) [inline virt-v2v/scripts/rhel/run/network_config_util.sh:170]: worth adding a log message?

File: virt-v2v/scripts/rhel/run/network_config_util.sh (hunk 1 above)
Comment (yaacov) [reply virt-v2v/scripts/rhel/run/network_config_util.sh]: thanks ! replaced ...

File: virt-v2v/scripts/rhel/run/network_config_util.sh (hunk 3 above)
Comment (yaacov) [reply virt-v2v/scripts/rhel/run/network_config_util.sh:163]: added logs for invalid mac to ip lines

File: virt-v2v/scripts/rhel/run/network_config_util.sh (hunk 4 above)
Comment (yaacov) [reply virt-v2v/scripts/rhel/run/network_config_util.sh:172]: added logs for device not found

File: virt-v2v/scripts/rhel/run/network_config_util.sh (hunk 2 above)
Comment (yaacov) [reply virt-v2v/scripts/rhel/run/network_config_util.sh:148]: added return here

**Review by mnecas — APPROVED** [review]

**Pull request merged**